src/test/spec/oidc.rs

```diff
@@ -1,14 +1,27 @@
+use crate::{
```

Comment:
I decided to correct these two tests in this PR as a start on the rest of the prose tests. These tests need to assert the number of times the callback is called.... getting a shared Arc into closures with async blocks is nearly impossible. There does not appear to be a way to do it without the top level function here.

Reply:
The ownership here is gnarly, but it can be made to work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=437cdfd395260f80a01ddce572d97d52

The trick is that you need to `clone` the `Arc` _twice_: once to give an owned value to the `Fn`, and a second time within that callback to give an owned value to the `Future`.  That second `clone` is needed because the outer callback can be called multiple times, and each time it has to produce a future that owns the value, so each time it's called it has to re-clone the `Arc`.

This can be boiled down a little bit to remove the async aspect, which makes it easier to see what's going on: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5f8a27ba1ae567f9ba337489d6a49a40

This is also why the distinction between `Fn` and `FnOnce` exists; if the Rust compiler knows that the closure can be called at most once, it's fine with transferring ownership from the closure to its return value: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=44aeb2634aa8e87c43e86ea349106f6a

Reply:
Interesting. I thought I tried cloning twice and that was what was really shocking to me. I must be misremembering